packages/flutter/lib/src/rendering/list_wheel_viewport.dart

```diff
@@ -0,0 +1,562 @@
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+
+import 'package:flutter/foundation.dart';
+import 'package:vector_math/vector_math_64.dart' show Matrix4;
+
+import 'box.dart';
+import 'object.dart';
+import 'viewport.dart';
+import 'viewport_offset.dart';
+
+typedef double _ChildSizingFunction(RenderBox child);
+
+class _ListWheelParentData extends ContainerBoxParentData<RenderBox> { }
+
+/// Render, onto a wheel, a bigger sequential set of objects inside this viewport.
+///
+/// Takes a scrollable set of fixed sized [RenderBox]es and renders them
+/// sequentially along the scrolling axis.
+// Can be augmented later to have configurable axis directions.
+///
+/// It also starts with the first scrollable item in the center of the main axis
+/// and ends with the last scrollable item in the center of the main axis. This
+/// is in contrast to typical lists that starts with the first scrollable item
+/// at the start of the main axis and ends with the last scrollable item at the
+/// end of the main axis.
+///
+/// Instead of rendering its children on a flat plane, it renders them
+/// as if each child is broken into its own plane and that plane is
+/// perpendicularly fixed onto a cylinder which rotates along the scrolling
+/// axis.
+///
+/// This class works in 3 coordinate systems:
+///
+/// 1- The **scrollable layout coordinates**. This coordinate system is used to
+///    communicate with [ViewportOffset] and describes its children's abstract
+///    offset from the beginning of the scrollable list at (0.0, 0.0).
+///
+///    The list is scrollable from the start of the first child item to the
+///    start of the last child item.
+///
+///    Children's layout coordinates don't change as the viewport scrolls.
+///
+/// 2- The **untransformed plane's viewport painting coordinates**. Children are
+///    not painted in this coordinate system. It's an abstract intermediary used
+///    before transforming into the next cylindrical coordinate system.
+///
+///    This system is the **scrollable layout coordinates** translated by the
+///    scroll offset such that (0.0, 0.0) is the top left corner of the
+///    viewport.
+///
+///    Because the viewport is centered at the scrollable list's scroll offset
+///    instead of starting at the scroll offset, there are paintable children
+///    ~1/2 viewport length before and after the scroll offset instead of ~1
+///    viewport length after the scroll offset.
+///
+///    Children's visibility inclusion in the viewport is determined in this
+///    system regardless of the cylinder's properties such as [diameterRatio]
+///    or [perspective]. In other words, a 100px long viewport will always
+///    paint 10-11 visible 10px children if there are enough children in the
+///    viewport.
+///
+/// 3- The **transformed cylindrical space viewport painting coordinates**.
+///    Paintable children from system 2 get their positions transformed into
```

Comment:
What's a non-paintable child?

Unless this means something specific, I'd remove the word "Paintable".

Reply:
Done